query-node/mappings/src/common.ts

```diff
@@ -96,42 +96,105 @@ export async function prepareDataObject(
   return dataObject
 }
 
-export function isVideoFullyActive(video: Video): boolean {
-  return (
+export interface IVideoActiveStatus {
+  isFullyActive: boolean
+  videoCategory: VideoCategory | undefined
+  channel: Channel
+  channelCategory: ChannelCategory | undefined
+}
+
+export function getVideoActiveStatus(video: Video): IVideoActiveStatus {
+  const isFullyActive =
     !!video.isPublic &&
     !video.isCensored &&
     video.thumbnailPhotoAvailability === AssetAvailability.ACCEPTED &&
     video.mediaAvailability === AssetAvailability.ACCEPTED
-  )
+  const videoCategory = video.category
+  const channel = video.channel
+  const channelCategory = channel.category
+
+  return {
+    isFullyActive,
+    videoCategory,
+    channel,
+    channelCategory,
+  }
 }
 
 export async function updateVideoActiveCounters(
   store: DatabaseManager,
-  video: Video,
-  isFullyActive: boolean
+  initialActiveStatus: IVideoActiveStatus | undefined,
+  activeStatus: IVideoActiveStatus | undefined
 ): Promise<void> {
-  const counterChange = isFullyActive ? 1 : -1
+  // definition of generic type for Hydra DatabaseManager's methods
+  type EntityType<T> = {
+    new (...args: any[]): T
+  }
 
-  if (video.channel) {
-    const channel = video.channel
-    channel.activeVideosCounter += counterChange
+  async function updateSingleEntity<Entity extends VideoCategory | Channel>(
+    entity: Entity,
+    counterChange: number
+  ): Promise<void> {
+    entity.activeVideosCounter += counterChange
 
-    await store.save<Channel>(channel)
+    await store.save<EntityType<Entity>>(entity)
```

Comment:
This doesn't seem to add any value in terms of typesafety (I can still pass invalid object or even a number as argument and the compiler won't complain). I think you can just use: `store.save(entity)` instead.